src/manipulators/no_implicit_any_manipulator.ts

```diff
@@ -26,9 +39,391 @@ export class NoImplicitAnyManipulator extends Manipulator {
+    successor: AcceptedDeclaration
+  ): void {
+    // Get predecessor's type.
+    const calculatedType = this.toTypeList(predecessor.getType());
+
+    // If predecessor is type any, add predecessor's declaration to the dependency graph and modified declarations set
+    // and add the dependency into the graph.
+    if (calculatedType.some(type => type.isAny())) {
+      if (Node.isIdentifier(predecessor)) {
+        predecessor
+          .getSymbol()
+          ?.getDeclarations()
+          ?.forEach(assignedDeclaration => {
+            if (
+              Node.isVariableDeclaration(assignedDeclaration) ||
+              Node.isParameterDeclaration(assignedDeclaration)
+            ) {
+              modifiedDeclarations.add(assignedDeclaration);
+              this.addToMapSet(dependencyGraph, assignedDeclaration, successor);
+            }
+          });
+      }
+    } else {
+      // Otherwise, only add predecessor's type to successor's calculated types.
+      calculatedType.forEach(type => {
+        this.addToMapSet(calculatedDeclarationTypes, successor, type);
+      });
+    }
+  }
+
+  /**
+   * Sets declaration types based on calculated declaration types.
+   * @param {Map<AcceptedDeclaration, Set<Type>>} calculatedDeclarationTypes - Calculated types for each declaration.
+   */
+  private setDeclarationTypes(
+    calculatedDeclarationTypes: Map<AcceptedDeclaration, Set<Type>>
+  ): void {
+    for (const [declaration, types] of calculatedDeclarationTypes) {
+      // Set declaration type.
+      const newDeclaration = declaration.setType(
+        Array.from(types)
+          .map(type => type.getText(declaration))
+          .sort()
+          .join(' | ')
+      );
+
+      // Add comment before edited declaration.
+      const modifiedStatement = this.getModifiedStatement(newDeclaration);
+      if (
+        modifiedStatement &&
+        this.verifyCommentRange(modifiedStatement, NO_IMPLICIT_ANY_COMMENT)
+      ) {
+        modifiedStatement.replaceWithText(
+          `${NO_IMPLICIT_ANY_COMMENT}\n${modifiedStatement
+            .getText()
+            .trimLeft()}`
+        );
+      }
+    }
+  }
+
+  /**
+   * Calculates final inferred declaration types in topological order based on declaration dependencies.
+   * @param {Map<AcceptedDeclaration, Set<Type>>} calculatedDeclarationTypes - Calculated types for each declaration.
+   * @param {AcceptedDeclaration[]} sortedDeclarations - Topologically sorted list of declarations.
+   * @param {Map<AcceptedDeclaration, Set<AcceptedDeclaration>>} dependencyGraph - Dependency graph between declarations.
+   */
+  private calculateDeclarationTypes(
+    calculatedDeclarationTypes: Map<AcceptedDeclaration, Set<Type>>,
+    sortedDeclarations: AcceptedDeclaration[],
+    dependencyGraph: Map<AcceptedDeclaration, Set<AcceptedDeclaration>>
+  ) {
+    // Runs code block twice to handle cyclical dependencies between declarations.
+    _.times(2, () => {
```

Comment:
let's construct a Map<node, set<node>>.

Reply:
Reimplemented `calculateDeclarationTypes` using the descendants map as we discussed.